src/transport/mod.rs

```diff
@@ -79,36 +72,64 @@ impl Transport {
         &mut self,
         target: &SocketAddr,
         content: Bytes,
-        timer_token: u64,
+        timer: &Timer,
+        delay: Duration,
     ) {
         let token = self.next_msg_token();
-        self.send_later(*target, content, token, timer_token);
+        self.send_later(*target, content, token, timer, delay);
         self.cache.insert_message(token, slice::from_ref(target), 1);
     }
 
-    pub fn target_failed(&mut self, msg_token: Token, failed_target: SocketAddr) -> Resend {
-        self.cache.target_failed(msg_token, failed_target)
+    pub fn send_message_to_client(&mut self, target: SocketAddr, msg: Bytes, token: Token) {
+        let client = Peer::Client(target);
+        self.quic_p2p.send(client, msg, token);
     }
 
-    pub fn send_now(&mut self, target: SocketAddr, content: Bytes, token: Token) {
-        self.quic_p2p.send(Peer::Node(target), content, token)
+    pub fn target_succeeded(&mut self, token: Token, target: SocketAddr) {
+        self.cache.target_succeeded(token, target)
     }
 
-    pub fn send_later(
+    pub fn target_failed(
```

Comment:
This is a neat and tidy method now. It's telling a lot about what is happening, with no more than its own responsibilities.

The code is starting to tell a story :) (In human speak)